packages/@lwc/engine-core/src/framework/rendering.ts

```diff
@@ -753,14 +753,17 @@ function allocateInSlot(vm: VM, children: VNodes, owner: VM) {
             continue;
         }
 
-        let slotName = '';
+        let slotName: string | number | boolean = '';
         if (isVBaseElement(vnode)) {
-            slotName = (vnode.data.attrs?.slot as string) ?? '';
+            slotName = vnode.data.attrs?.slot ?? '';
         } else if (isVScopedSlotFragment(vnode)) {
             slotName = vnode.slotName;
         }
 
-        const vnodes: VNodes = (cmpSlotsMapping[slotName] = cmpSlotsMapping[slotName] || []);
+        const normalizedSlotName = slotName.toString();
```

Comment:
```suggestion
        const normalizedSlotName = String(slotName);
```
Works in [IE11 too](https://caniuse.com/mdn-javascript_builtins_string_string)

Reply:
We also have our own `toString` function in `@lwc/shared`:

https://github.com/salesforce/lwc/blob/08684f3e73db5916be122f571fd15a4f7ddf80e8/packages/@lwc/shared/src/language.ts#L139